[bug] Slave is not configured or failed to initialize properly.

**Describe the problem**

Initialization error, But the cluster can be started normally, elections.

**To Reproduce**

```
init mysql log:
 2022-04-14T15:41:38.072702+08:00 1 [ERROR] 1794  Slave is not configured or failed to initialize properly. You must at least set --server-id to enable either a master or a slave. Additional error messages can be found in the MySQL error log.

 2022-04-14T15:41:38.073601+08:00 0 [ERROR] Aborting
```

**Expected behavior**


**Environment:**

- RadonDB MySQL version: v2.2.0

```
Initializing database
2022-04-14T16:02:09.471842+08:00 0 [Warning] TIMESTAMP with implicit DEFAULT value is deprecated. Please use --explicit_defaults_for_timestamp server option (see documentation for more details).
2022-04-14T16:02:09.471908+08:00 0 [Warning] 'NO_ZERO_DATE', 'NO_ZERO_IN_DATE' and 'ERROR_FOR_DIVISION_BY_ZERO' sql modes should be used with strict mode. They will be merged with strict mode in a future release.
2022-04-14T16:02:09.471917+08:00 0 [Warning] 'NO_AUTO_CREATE_USER' sql mode was not set.
 100 200 300 400 500 600 700 800 900 1000
 100 200 300 400 500 600 700 800 900 1000
2022-04-14T16:03:09.699276+08:00 0 [Warning] InnoDB: New log files created, LSN=45790
2022-04-14T16:03:10.086870+08:00 0 [Warning] InnoDB: Creating foreign key constraint system tables.
2022-04-14T16:03:10.157348+08:00 0 [Warning] unknown variable 'loose_tokudb_directio=ON'
2022-04-14T16:03:10.157407+08:00 0 [Warning] unknown variable 'loose_tokudb_directio=ON'
2022-04-14T16:03:10.161937+08:00 0 [Warning] No existing UUID has been found, so we assume that this is the first time that this server has been started. Generating a new UUID: 5346e840-bbc9-11ec-a07a-8a228d5caa56.
2022-04-14T16:03:10.163483+08:00 0 [Warning] Gtid table is not ready to be used. Table 'mysql.gtid_executed' cannot be opened.
2022-04-14T16:03:10.582890+08:00 1 [Warning] root@localhost is created with an empty password ! Please consider switching off the --initialize-insecure option.
2022-04-14T16:03:12.291097+08:00 1 [ERROR] 1794  Slave is not configured or failed to initialize properly. You must at least set --server-id to enable either a master or a slave. Additional error messages can be found in the MySQL error log.
2022-04-14T16:03:12.292255+08:00 0 [ERROR] Aborting
```

--- a/sidecar/config.go
+++ b/sidecar/config.go
@@ -359,7 +359,7 @@ func (cfg *Config) buildXenonConf() []byte {
 }
 
 // buildInitSql used to build init.sql. The file run after the mysql init.
-func (cfg *Config) buildInitSql() []byte {
+func (cfg *Config) buildInitSql(hasInit bool) []byte {
 	sql := fmt.Sprintf(`SET @@SESSION.SQL_LOG_BIN=0;
 CREATE DATABASE IF NOT EXISTS %s;
 DROP user IF EXISTS 'root'@'127.0.0.1';
@@ -381,7 +381,7 @@ DROP user IF EXISTS '%s'@'%%';
 CREATE USER '%s'@'%%' IDENTIFIED BY '%s';
 GRANT ALL ON %s.* TO '%s'@'%%' ;
 FLUSH PRIVILEGES;
-RESET SLAVE ALL;
+
 `,
 		cfg.Database, //database
 		cfg.RootPassword,
@@ -401,6 +401,10 @@ RESET SLAVE ALL;
 		cfg.User,               //drop user
 		cfg.User, cfg.Password, //create user
 		cfg.Database, cfg.User) //grant
+
+	if hasInit {
+		sql += "\nRESET SLAVE ALL;\n"
+	}
 	return utils.StringToBytes(sql)
 }
 

--- a/sidecar/init.go
+++ b/sidecar/init.go
@@ -168,12 +168,9 @@ func runInitCommand(cfg *Config) error {
 	if err = ioutil.WriteFile(initFilePath+"/reset.sql", []byte("reset master;"), 0644); err != nil {
 		return fmt.Errorf("failed to write reset.sql: %s", err)
 	}
-
+	hasInitialized, _ := checkIfPathExists(path.Join(dataPath, "mysql"))
 	// build init.sql.
 	initSqlPath := path.Join(extraConfPath, "init.sql")
-	if err = ioutil.WriteFile(initSqlPath, cfg.buildInitSql(), 0644); err != nil {
-		return fmt.Errorf("failed to write init.sql: %s", err)
-	}
 
 	// build extra.cnf.
 	extraConfig, err := cfg.buildExtraConfig(initSqlPath)
@@ -183,7 +180,7 @@ func runInitCommand(cfg *Config) error {
 
 	// Notice: plugin.cnf cannot be copied to /etc/mysql/conf.d when initialized.
 	// Check /var/lib/mysql/mysql exists. if exists it means that been initialized.
-	if exists, _ := checkIfPathExists(path.Join(dataPath, "mysql")); exists || strings.HasPrefix(getEnvValue("MYSQL_VERSION"), "5") {
+	if hasInitialized || strings.HasPrefix(getEnvValue("MYSQL_VERSION"), "5") {
 		// Save plugin.cnf and extra.cnf to /etc/mysql/conf.d.
 		saveCnfTo(extraConfPath, extraConfig)
 	} else {
@@ -228,7 +225,7 @@ func runInitCommand(cfg *Config) error {
 	// Check datadir is empty.
 	// if /var/lib/mysql/mysql is empty, then run the restore.
 	// otherwise , it must be has data, then do nothing.
-	if exists, _ := checkIfPathExists(dataPath + "/mysql"); !exists {
+	if !hasInitialized {
 		if len(cfg.XRestoreFrom) != 0 {
 			var err_f error
 			if cfg.CloneFlag {
@@ -241,10 +238,14 @@ func runInitCommand(cfg *Config) error {
 					return fmt.Errorf("failed to restore from %s: %s", cfg.XRestoreFrom, err)
 				}
 			}
-
+			// Check has initialized again.
+			hasInitialized, _ = checkIfPathExists(path.Join(dataPath, "mysql"))
 		}
 	}
-
+	// Build init.sql after restore
+	if err = ioutil.WriteFile(initSqlPath, cfg.buildInitSql(hasInitialized), 0644); err != nil {
+		return fmt.Errorf("failed to write init.sql: %s", err)
+	}
 	// build xenon.json.
 	xenonFilePath := path.Join(xenonPath, "xenon.json")
 	if err = ioutil.WriteFile(xenonFilePath, cfg.buildXenonConf(), 0644); err != nil {

